File: rbp_flutter/lib/data/models/debt.dart
class Debt {
  const Debt({
    this.id,
    required this.userId,
    required this.name,
    required this.principalAmount,
    required this.annualRate,
    required this.termMonths,
    required this.startDate,
    required this.paymentDay,
    required this.monthlyPayment,
    required this.currentBalance,
    this.isActive = 1,
    this.createdAt,
    this.updatedAt,
  });

  final int? id;
  final int userId;
  final String name;
  final double principalAmount;
  final double annualRate;
  final int termMonths;
  final String startDate;
  final int paymentDay;
  final double monthlyPayment;
  final double currentBalance;
  final int isActive;
  final String? createdAt;
  final String? updatedAt;

  bool get isActiveBool => isActive == 1;

  factory Debt.fromMap(Map<String, Object?> map) {
    return Debt(
      id: map['id'] as int?,
      userId: (map['user_id'] as num).toInt(),
      name: (map['name'] ?? '') as String,
      principalAmount: (map['principal_amount'] as num).toDouble(),
      annualRate: (map['annual_rate'] as num).toDouble(),
      termMonths: (map['term_months'] as num).toInt(),
      startDate: (map['start_date'] ?? '') as String,
      paymentDay: (map['payment_day'] as num).toInt(),
      monthlyPayment: (map['monthly_payment'] as num).toDouble(),
      currentBalance: (map['current_balance'] as num).toDouble(),
      isActive: (map['is_active'] as num?)?.toInt() ?? 1,
      createdAt: map['created_at'] as String?,
      updatedAt: map['updated_at'] as String?,
    );
  }
}

File: backend/schemas/debt.py
from __future__ import annotations

from datetime import date as date_cls

from pydantic import BaseModel, ConfigDict, Field


class DebtCreate(BaseModel):
    name: str = Field(min_length=1, max_length=160)
    principal_amount: float = Field(gt=0)
    annual_rate: float = Field(ge=0)
    term_months: int = Field(gt=0, le=600)
    start_date: date_cls
    payment_day: int = Field(ge=1, le=31)


class DebtUpdate(DebtCreate):
    pass


class DebtPaymentCreate(BaseModel):
    payment_date: date_cls
    total_amount: float = Field(gt=0)
    interest_amount: float = Field(ge=0)
    capital_amount: float = Field(gt=0)
    notes: str | None = None


class DebtPaymentRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    debt_id: int
    payment_date: date_cls
    total_amount: float
    interest_amount: float
    capital_amount: float
    notes: str | None = None


class DebtRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    user_id: int
    name: str
    principal_amount: float
    annual_rate: float
    term_months: int
    start_date: date_cls
    payment_day: int
    monthly_payment: float
    current_balance: float
    is_active: bool


class DebtSummary(BaseModel):
    count: int
    total_balance: float
    monthly_payment_total: float


class DebtListResponse(BaseModel):
    items: list[DebtRead]
    summary: DebtSummary


class PersonalDebtCreate(BaseModel):
    person: str = Field(min_length=1, max_length=160)
    total_amount: float = Field(gt=0)
    description: str | None = Field(default=None, max_length=255)
    date: date_cls


class PersonalDebtUpdate(PersonalDebtCreate):
    pass


class PersonalDebtPaymentCreate(BaseModel):
    payment_date: date_cls
    amount: float = Field(gt=0)
    notes: str | None = None


class PersonalDebtPaymentRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    personal_debt_id: int
    payment_date: date_cls
    amount: float
    notes: str | None = None


class PersonalDebtRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    user_id: int
    person: str
    total_amount: float
    current_balance: float
    description: str | None = None
    date: date_cls
    is_paid: bool
    paid_date: date_cls | None = None


class PersonalDebtSummary(BaseModel):
    count: int
    total_balance: float


class PersonalDebtListResponse(BaseModel):
    items: list[PersonalDebtRead]
    summary: PersonalDebtSummary
